R/model.R

```diff
@@ -146,220 +160,368 @@ fall_run_model <- function(scenario = NULL, seeds = NULL){
+            rbinom(n = 4, size = round(yearlings_at_south_delta[i, ]), prob = migratory_survival$delta[4, ])
+          }))
+          
+          # migrants.out<-rbinMatVector(N.delt.fsh$out2ocean,Sac.Delt.S[1,],stochastic)
+          yearlings_out <- t(sapply(1:nrow(yearlings_at_north_delta), function(i) {
+            rbinom(n = 4, 
+                   size = round(yearlings_at_north_delta[i, ]), 
+                   prob = migratory_survival$sac_delta[1, ])
+          }))
+          
+          
+          # migrants.at.GG<-rbinMatObject(migrants.out,Bay.S,stochastic)+rbinMatObject(holdSdelta,Bay.S,stochastic)
+          survived_yearlings_out <- t(sapply(1:nrow(yearlings_out), function(i) {
+            rbinom(n = 4,
+                   size = round(yearlings_out[i, ]),
+                   prob = migratory_survival$bay_delta)
+          }))
+          
+          survived_yearling_holding_south_delta <- t(sapply(1:nrow(yearling_holding_south_delta), function(i) {
+            rbinom(n = 4,
+                   size = round(yearling_holding_south_delta[i, ]),
+                   prob = migratory_survival$bay_delta)
+          }))
+          
+          yearlings_at_golden_gate <- survived_yearlings_out + survived_yearling_holding_south_delta
+          
+          
+          juveniles_at_chipps <- juveniles_at_chipps + yearlings_out + yearling_holding_south_delta
+          
+          adults_in_ocean <- adults_in_ocean + ocean_entry_success(migrants = yearlings_at_golden_gate,
+                                                                   month = 11,
+                                                                   avg_ocean_transition_month = avg_ocean_transition_month)
+        }
+        
         # if month < 8
         # route northern natal fish stay and rear or migrate downstream ------
-        upper_sac_trib_fish <-  route(year = year,
+        upper_sac_trib_fish <-  route(year = juv_dynamics_year,
                                       month = month,
                                       juveniles = juveniles[1:15, ],
                                       inchannel_habitat = habitat$inchannel[1:15],
                                       floodplain_habitat = habitat$floodplain[1:15],
                                       prop_pulse_flows =  prop_pulse_flows[1:15, ],
                                       detour = 'sutter')
-
+        
         upper_sac_trib_rear <- rear(juveniles = upper_sac_trib_fish$inchannel,
                                     survival_rate = rearing_survival$inchannel[1:15, ],
                                     growth = growth_rates,
                                     floodplain_juveniles = upper_sac_trib_fish$floodplain,
                                     floodplain_survival_rate = rearing_survival$floodplain[1:15, ],
                                     floodplain_growth = growth_rates_floodplain,
                                     weeks_flooded = weeks_flooded[1:15, month, year])
```

Comment:
For all the rear functions do we need to use juv_dynamics_year? Year is used in the `weeks_flooded` input. `weeks_flooded[23, month, year]` I think that these years should be replaced with `juv_dynamics_year` in this rear and all the `rear()` below

Reply:
good catch! yeah everything that references a year within the month iteration should be using the juv_dynamics_year